functions/wait/action/kubernetesUpgrade.go

```diff
@@ -17,32 +17,44 @@ func KubernetesUpgrade(t *testing.T, client *rancher.Client, clusterID string, m
 
 	if module == "aks" || module == "gke" {
 		expectedUpgradedKubernetesVersion := `v` + clusterConfig.UpgradedKubernetesVersion
-		err := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
-			cluster, err := client.Management.Cluster.ByID(clusterID)
+		waiting := false
+		wait_err := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
+			cluster, client_err := client.Management.Cluster.ByID(clusterID)
 			require.NoError(t, err)
 
-			if err != nil {
+			if client_err != nil {
 				t.Logf("Failed to locate cluster and grab client specs. Error: %v", err)
 				return false, err
 			}
 
 			if cluster.Version.GitVersion == expectedUpgradedKubernetesVersion {
+				t.Logf("Successfully updated kubernetes version to %v", clusterConfig.UpgradedKubernetesVersion)
 				return true, nil
 			}
 
-			t.Logf("Failed to instantiate wait poll.")
+			if !waiting {
```

Comment:
why not just move this outside the poll, so that there's no need for the if statement?

Reply:
If the poll fails to instantiate, i don't want a log to misinform the user that the poll is waiting.